tree-sitter-stack-graphs/src/cli/analyze.rs

```diff
@@ -192,4 +236,19 @@ impl AnalyzeArgs {
         file_status.ok("success")?;
         Ok(())
     }
+
+    fn skip(&self, path: &Path, seen_mark: &mut bool) -> bool {
+        if *seen_mark {
+            false
+        } else if let Some(mark) = &self.continue_from {
+            if path != mark {
+                true
+            } else {
+                *seen_mark = true;
+                false
+            }
+        } else {
+            false
+        }
+    }
```

Comment:
This function is a little cryptic. Could you add some docs? It’s not clear what the return type represents. I also think you could make it a little more readable by having a `return false` clause at the end of the function and moving the bulk of the logic (lines 244-248) into one if-clause guarded by `if !*seen_mark && let Some(mark…)` (assuming you can in fact mix let and if-let in a conditional).

Reply:
Alas, mixing boolean and matching in an if-clause is not allowed. It would cleanup so nicely sometimes...